src/blocks/scratchpad.rs

```diff
@@ -0,0 +1,86 @@
+//! Scratchpad indicator
+//!
+//! # Configuration
+//!
+//! Key | Values | Default
+//! ----|--------|--------
+//! `format`          | A string to customise the output of this block | ` $icon $count.eng(range:1..) |`
+//!
+//! Placeholder | Value                                      | Type   | Unit
+//! ------------|--------------------------------------------|--------|-----
+//! `icon`      | A static icon                              | Icon   | -
+//! `count`     | Number of windows in scratchpad            | Number | -
+//!
+//! # Example
+//!
+//! ```toml
+//! [[block]]
+//! block = "scratchpad"
+//!
+//! # Icons Used
+//! - `scratchpad`
+
+use swayipc_async::{Connection, Event as SwayEvent, EventType, Node, WindowChange};
+
+use super::prelude::*;
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(deny_unknown_fields, default)]
+pub struct Config {
+    pub format: FormatConfig,
+}
+
+fn count_scratchpad_windows(node: &Node) -> usize {
+    node.find_as_ref(|n| n.name == Some("__i3_scratch".to_string()))
+        .map(|node| node.floating_nodes.len())
+        .unwrap_or(0)
```

Comment:
To not allocate a new string on each iteration:

```suggestion
    node.find_as_ref(|n| n.name.as_deref() == Some("__i3_scratch"))
        .map_or(0, |node| node.floating_nodes.len())
```